mlrun/projects/project.py

```diff
@@ -3446,6 +3448,85 @@ def remove_custom_packager(self, packager: str):
         """
         self.spec.remove_custom_packager(packager=packager)
 
+    def create_api_gateway(
+        self,
+        name: str,
+        host: str = None,
+        path: str = "",
+        description: str = "",
+        functions: Union[
+            list[Union[RemoteRuntime, ServingRuntime]],
+            Union[RemoteRuntime, ServingRuntime],
+        ] = [],
```

Comment:
And, I see no validation on projects. Specifically, a GW belongs to a project. Can it sit in front of functions from different projects? If not (which I believe it shouldn't), then you should also validate that the functions you get are from the same project as the one for which you're creating the GW for.

Reply:
@theSaarco that's a very good point, will add validation!

Reply:
@rokatyy - just as a note, I'm not 100% sure we should block putting a gw in front of cross-project objects, but this needs to be properly considered before we allow such a thing (and as 1st stage implementation it makes sense to block).